include/openthread/platform/radio.h

```diff
@@ -159,6 +159,29 @@ struct otExtAddress
  */
 typedef struct otExtAddress otExtAddress;
 
+/**
+ * Converts an array representing an IEEE 802.15.4 Extended Address into a
+ * 64-bit value.
+ *
+ * This is necessary as the otExtAddress array is not necessary aligned for
+ * access as a 64-bit value.
+ */
+static inline uint64_t otExtAddressLoadUint64(const otExtAddress *address)
```

Comment:
AFAICT, this function is only needed by the TI platforms, currently. Can we move this to TI-specific platform code?

Reply:
I'm not against that idea. Technically it could be needed by other platforms if they ever want to load the address into a 64-bit word, but that is not the case at the moment.

However I would like to ask that someone else take over the pull request in that case. I don't see one location where I should put this function so that it will be TI-specific and still follow the overall structure of the openthread repo. I foresee some bikeshedding around the best location for the function and I don't know the project well enough to have a strong opinion.

Reply:
@srickardti, thoughts?

Reply:
@srickardti , I agree with @jwhui in that I feel this function does not really belong in `radio.h`. OT radio platform definition treat the extended MAC address as an array of 8 bytes in `otExtAddress`.

To me, this is more of a generic helper function that reads the value from a pointer (to an array/buffer) as a `uint64_t` with a specific encoding (little endian). 

- In core we have `common/encoding.hpp` which provides different `ReadUint16/ReadUint32` flavors (for big/little endian).
- How about adding this function (with maybe other similar functions for reading `uint16_t` and `uint32_t`) for all platform layer code under `example/platforms/utils` folder in a `encoding.h` header (this way it can shared/use by all radios/platform, similar to how `utils/code_utils.h` header is used).

Reply:
@richarddonkin, @srickardti, @abtink - I updated this PR as @abtink suggested above. Please review.